fleetdb/methods.go

```diff
@@ -9,26 +9,85 @@ import (
 	"github.com/pkg/errors"
 )
 
+var (
+	errBadVAttr = errors.New("unable to deserialize versioned attribute")
+	errBadAttr  = errors.New("unable to deserialize attribute")
+)
+
 func UnpackVersionedAttribute(attr *ss.VersionedAttributes, dst any) error {
-	derr := errors.New("error deserializing versioned attribute")
 	err := json.Unmarshal(attr.Data, dst)
 	if err != nil {
-		return errors.Wrap(derr, err.Error())
+		return errors.Wrap(errBadVAttr, err.Error())
 	}
 
 	return nil
 }
 
 func UnpackAttribute(attr *ss.Attributes, dst any) error {
-	derr := errors.New("error deserializing attribute")
 	err := json.Unmarshal(attr.Data, dst)
 	if err != nil {
-		return errors.Wrap(derr, err.Error())
+		return errors.Wrap(errBadAttr, err.Error())
 	}
 
 	return nil
 }
 
+// RecordToComponent takes a single incoming FleetDB component record and creates
+// a rivets Component from it. An important difference from ConvertComponents (cf.
+// below) is that this sets a given attribute/variable attribute preferentially
+// from in-band data, using out-of-band only when there is no in-band inventory.
+// We find that in-band inventory data is more complete, especially for SuperMicro.
+// The only attributes returned here are firmware-version, and status. We skip
+// reporting metadata (often a long list of capabilities).
```

Comment:
It would be worth including capabilities (at the least for NICs, Drives) since a bunch of effort went into collecting that data and it helps us determine things like - if a drive supports secure erase.
cc @luke-jarymowycz @psteinbachs

Reply:
My initial thought was to skip that but I think that actually conflates this routine with the purposes of the caller. I think it's up to the caller to determine what set of facets they return to their clients. I don't think the `metadata` namespace is particularly useful either, but that also is tangential to this code.